src/shell-integration/nushell/vendor/autoload/ghostty-integration.nu

```diff
@@ -0,0 +1,114 @@
+# Enables SSH environment variable compatibility.
+# Converts TERM from xterm-ghostty to xterm-256color
+# and propagates COLORTERM, TERM_PROGRAM, and TERM_PROGRAM_VERSION
+# check your sshd_config on remote host to see if these variables are accepted
+def set_ssh_env []: nothing -> record<ssh_term: string, ssh_opts: list<string>> {
+    return {ssh_term: "xterm-256color", ssh_opts: ["-o", "SetEnv COLORTERM=truecolor", "-o", "SendEnv TERM_PROGRAM TERM_PROGRAM_VERSION"]}
+}
+
+# Enables automatic terminfo installation on remote hosts.
+# Attempts to install Ghostty's terminfo entry using infocmp and tic when
+# connecting to hosts that lack it. 
+# Requires infocmp to be available locally and tic to be available on remote hosts.
+# Caches installations to avoid repeat installations.
+def set_ssh_terminfo [ssh_opts: list<string>, ssh_args: list<string>] {
+  mut ssh_opts = $ssh_opts
+  let ssh_cfg = ^ssh -G ...($ssh_args)
+    | lines 
+    | split column -n 2 " " key value
+    | where key == "user" or key == "hostname"
+    | transpose -r
+    | into record
```

Comment:
I think we can improve this record-building pipeline a little bit:


```suggestion
  let ssh_cfg = ^ssh -G ...($ssh_args)
    | lines 
    | parse "{key} {value}"
    | where key in ["user", "hostname"]
    | select key value
    | transpose -rd
    | into record
    | default { user: $env.USER, hostname: "localhost" }
```

1. Use `parse` rather than `split`
2. Use `transpose -rd` to automatically handle the header row
3. Use `where key in ...` for readability and extensibility
4. Declare defaults "just in case", which nushell makes easy